sbt_common/internal_model/src/main/scala/uk/ac/wellcome/models/utils/CSVReferenceTypeManager.scala

```diff
@@ -0,0 +1,53 @@
+package uk.ac.wellcome.models.utils
+
+import java.io.InputStream
+
+import uk.ac.wellcome.exceptions.GracefulFailureException
+
+import scala.io.Source
+
+class CSVReferenceTypeManager(path: String) {
+  private val stream: InputStream = getClass.getResourceAsStream(path)
+
+  // Our reference CSVs are files with three columns per row:
+  //
+  //    platformId,id,label
+  //
+  // For example:
+  //
+  //    2,acqi,Info Service acquisitions
+  //    3,acql,Wellcome Library
+  //    4,admi,administration
+  //
+  // The first entry is an identifier we can use to identify a row
+  // within a reference file.  The second and third entries are the ID
+  // and label we show in public ID schemes.
+  //
+  private val csvRows: Array[(String, String, String)] = Source
+    .fromInputStream(stream)
```

Comment:
Should we hold the desired resource path, and create the stream where used?
`private val resourcePath = path`
`Try(Source.fromResource(resource)...`